[quant][fix] Add quant_min/quant_max for default dynamic quantization observer

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89272
* #89271
* #89270
* #89269
* #89268
* #89248
* __->__ #89267

Summary:
This is needed for choose qparams, but previously it is not configurable, and in the reference quantization flow
with decomposed Tensor, we are making this explicit

Test Plan:
tested in future PR

Reviewers:

Subscribers:

Tasks:

Tags:

cc @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel

jerryzh168 committed:
[quant][fix] Add quant_min/quant_max for default dynamic quantization observer

Summary:
This is needed for choose qparams, but previously it is not configurable, and in the reference quantization flow
with decomposed Tensor, we are making this explicit

Test Plan:
tested in future PR

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix] Add quant_min/quant_max for default dynamic quantization observer"


Summary:
This is needed for choose qparams, but previously it is not configurable, and in the reference quantization flow
with decomposed Tensor, we are making this explicit

Test Plan:
tested in future PR

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix] Add quant_min/quant_max for default dynamic quantization observer"


Summary:
This is needed for choose qparams, but previously it is not configurable, and in the reference quantization flow
with decomposed Tensor, we are making this explicit

Test Plan:
tested in future PR

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix] Add quant_min/quant_max for default dynamic quantization observer"


Summary:
This is needed for choose qparams, but previously it is not configurable, and in the reference quantization flow
with decomposed Tensor, we are making this explicit

Test Plan:
tested in future PR

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on torch/ao/quantization/observer.py:
```diff
@@ -1316,6 +1316,8 @@ class PlaceholderObserver(ObserverBase):
     Args:
         dtype: dtype argument to the `quantize` node needed to implement the
                reference model spec.
+        quant_min: minimum value in quantized domain (TODO: align behavior with other observers)
```
what does the TODO mean?  Ideally if it's something important it would be done in the same PR, since observers are widely used with BC guarantees.

jerryzh168 replied:
it means removing this one: https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/observer.py#L240-L241 it will be a bigger change and need to handle BC etc. we need to have a rule to say when quant_min/quant_max should be specified I think

jerryzh168 replied:
I'll land for now to unblock the work, I think we will need to fix the tech debts a bit later..

jerryzh168 commented:
@pytorchbot merge -g

jerryzh168 committed:
Update on "[quant][fix] Add quant_min/quant_max for default dynamic quantization observer"


Summary:
This is needed for choose qparams, but previously it is not configurable, and in the reference quantization flow
with decomposed Tensor, we are making this explicit

Test Plan:
tested in future PR

Reviewers:

Subscribers:

Tasks:

Tags:

cc jianyuh raghuramank100 jamesr66a vkuzo jgong5 Xia-Weiwen leslie-fang-intel

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot merge
